docker-compose/datasets/Dockerfile

```diff
@@ -26,7 +27,11 @@ RUN apt-get update -qq \
     gzip \
     locales \
     locales-all \
-    wget
+    wget \
+    apt-transport-https \
+    gpg \
+    adoptium-ca-certificates \
+    temurin-21-jdk
```

Comment:
(note: these comments apply to all images)

you should install `temurin-21-jre` rather than `temurin-21-jdk` (we only need the JRE in the docker images)

`openssl` and `adoptium-ca-certificates` are already listed as dependencies of `temurin-21-(jre|jdk)`, there is no need to list them here (unless we exclicitly use them, but i am pretty sure we don't).

I would rather suggest to delay this PR until #2209 is merged because it will highly conflict with it (also #2209 makes a point in *not* installing build time dependencies (like wget, gpg or jdk) in the final images).

Reply:
Okay, we'll wait for your PR to be merged and I'll rebase myself on it. Thanks !